tools/flash_shelly.py

```diff
@@ -0,0 +1,499 @@
+#   -h                         This help text.
+#
+#  usage: python3 flash_shelly.py -la
+#  usage: python3 flash_shelly.py shelly1-034FFF
+"""
+
+logging.basicConfig(format='%(message)s')
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.WARNING)
+
+WHITE='\033[1m' # Bright White
+RED='\033[1;91m' # Bright Purple
+GREEN='\033[1;92m' # Bright Green
+YELLOW='\033[1;93m' # Bright Yellow
+BLUE='\033[1;94m' # Bright Blue
+PURPLE='\033[1;95m' # Bright Purple
+NC='\033[0m' # Normal Color
+
+arch = platform.system()
+
+if not importlib.util.find_spec("zeroconf"):
+  print('Installing zeroconf...')
+  pipe = Popen('pip3 install zeroconf', shell=True, stdout=PIPE).stdout
+  out = (pipe.read())
+  from zeroconf import ServiceBrowser, Zeroconf
+else:
+  from zeroconf import ServiceBrowser, Zeroconf
+
+if not importlib.util.find_spec("requests"):
+  print('Installing requests...')
+  pipe = Popen('pip3 install requests', shell=True, stdout=PIPE).stdout
+  out = (pipe.read())
+  import requests
+else:
+  import requests
+
+if not importlib.util.find_spec("packaging"):
+  print('Installing packaging...')
+  pipe = Popen('pip3 install packaging', shell=True, stdout=PIPE).stdout
+  out = (pipe.read())
+  from packaging import version
+else:
+  from packaging import version
+
+class MyListener:
+  global device_list, p_list
+  device_list=[]
+  p_list=[]
+
+  def add_service(self, zeroconf, type, name):
+      device_list.append(name.replace('._http._tcp.local.',''))
+      # info = zeroconf.get_service_info(type, name, 2000)
+      # logger.debug('INFO: %s' % info)
+      # properties = { y.decode('ascii'): info.properties.get(y).decode('ascii') for y in info.properties.keys() }
+      # p_list.append(properties)
+      # logger.debug('properties: %s' % properties)
+      # json_object = json.dumps(properties, indent = 2)
+
+def shelly_model(type, mode):
+  if mode != 'stock':
+    options = {'SHSW-1' : 'Shelly1',
+               'switch1' : 'Shelly1',
+               'SHSW-PM' : 'Shelly1PM',
+               'switch1pm' : 'Shelly1PM',
+               'SHSW-21' : 'Shelly2',
+               'switch2' : 'Shelly2',
+               'SHSW-25' : 'Shelly25',
+               'switch25' : 'Shelly25',
+               'SHPLG-S' : 'ShellyPlugS',
+               'shelly-plug-s' : 'ShellyPlugS',
+               'SHDM-1' : 'ShellyDimmer',
+               'dimmer1' : 'ShellyDimmer',
+               'SHRGBW2' : 'ShellyRGBW2',
+               'rgbw2' : 'ShellyRGBW2',
+    }
+  else:
+    options = {'switch1' : 'SHSW-1',
+               'switch1pm' : 'SHSW-PM',
+               'switch2' : 'SHSW-21',
+               'switch25' : 'SHSW-25',
+               'shelly-plug-s' : 'SHPLG-S',
+               'dimmer1' : 'SHDM-1',
+               'rgbw2' : 'SHRGBW2',
+    }
+  return(options[type])
+
+def write_flash(device, lfw, dlurl, cfw_type, mode):
+  logger.info("\n" + WHITE + "write_flash" + NC)
```

Comment:
you use both logger and prints for diagnostic output, use one or the other. my preference is logger.

Reply:
you can't, that why logging states use print for normal running code.

Reply:
I use 2 levels of debug, we only have `info` and `debug` to use, so leaves nothing for standard logging.

Reply:
https://docs.python.org/3/howto/logging.html

it states the following, which is what I am doing.
![Screenshot 2020-10-07 at 01 03 50](https://user-images.githubusercontent.com/77936/95272708-f8c84f80-0838-11eb-86e9-bfc7d7e42d0e.jpg)

Reply:
ok, I have managed to add a custom log level, tomorrow I'll replace print() with logging.

Reply:
done https://github.com/mongoose-os-apps/shelly-homekit/pull/176/commits/d444400fdda05dc1ef91777762dfa621309da183